operator/v1/types_ingress.go

```diff
@@ -1684,3 +1699,50 @@ type IngressControllerList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []IngressController `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:object:root=true
+//
+// Ingress contains configuration options specific to the Ingress Operator itself.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Ingress struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the Ingress Operator.
+	Spec IngressSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the Ingress Operator.
+	Status IngressStatus `json:"status,omitempty"`
+}
+
+type IngressSpec struct {
+	// operatorLogLevel is the log level of the ingress operator. Options are:
+	// * Normal
+	// * Debug
+	// * Trace
+	// * TraceAll
+	// When unspecified, the default level is "Normal".
+	//
+	// +optional
+	OperatorLogLevel LogLevel `json:"operatorLogLevel,omitempty"`
+}
+
+type IngressStatus struct{}
```

Comment:
I added a status field to this new type since it seems like it's expected, but as of right now, it's empty. Would it be better to not define it at all, and add it when necessary?

Reply:
I think it would be good to keep it even though it's empty, we should mark it as a subresource to make sure operators can update it without touching the spec

(We should double check what other operators have in their spec, might find some inspiration)